worlds/pokemon_emerald/rom.py

```diff
@@ -840,18 +842,26 @@ def _randomize_move_tutor_moves(world: "PokemonEmeraldWorld", patch: PokemonEmer
     # Always set Fortree move tutor to Dig
     patch.write_token(
         APTokenTypes.WRITE,
-        data.rom_addresses["gTutorMoves"] + (24 * 2),
+        data.rom_addresses["gTutorMoves"] + (FORTREE_MOVE_TUTOR_INDEX * 2),
         struct.pack("<H", data.constants["MOVE_DIG"])
     )
 
     # Modify compatibility
     if world.options.tm_tutor_compatibility.value != -1:
         for species in data.species.values():
+            compatibility = bool_array_to_int([
+                world.random.randrange(0, 100) < world.options.tm_tutor_compatibility.value
+                for _ in range(32)
+            ])
+
+            # Make sure Dig tutor has reasonable (>50%) compatibility
```

Comment:
Minor clarity change
```suggestion
            # Make sure Dig tutor has reasonable (>=50%) compatibility
```